spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+  end
+
+  it 'renders data for the last 2 years when `last 2 years` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 2.years
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-2-years').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  def initial_page_stub
+    from = Time.zone.today - 30.days
+    to = Time.zone.today
+
+    content_data_api_has_metric(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
+        base_path: '/base/path',
+        unique_pageviews: 145_000,
+        pageviews: 200_000,
+        title: "Content Title",
+        first_published_at: '2018-02-01T00:00:00.000Z',
+        public_updated_at: '2018-04-25T00:00:00.000Z',
+        primary_organisation_title: 'The ministry',
+        document_type: "news_story"
+      })
+
+    content_data_api_has_timeseries(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
+        unique_pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 0 },
+          { "date" => (from - 2.days).to_s, "value" => 9 },
+          { "date" => (to + 1.day).to_s, "value" => 9 }
+        ],
+        pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 8 },
+          { "date" => (from - 2.days).to_s, "value" => 8 },
+          { "date" => (to + 1.day).to_s, "value" => 8 }
+        ]
+      })
+  end
+
+  def stub_for(from, to)
+    content_data_api_has_metric(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
+        base_path: '/base/path',
+        unique_pageviews: 145_000,
+        pageviews: 200_000,
+        title: "Content Title",
+        first_published_at: '2018-02-01T00:00:00.000Z',
+        public_updated_at: '2018-04-25T00:00:00.000Z',
+        primary_organisation_title: 'The ministry',
+        document_type: "news_story"
+      })
+
+    content_data_api_has_timeseries(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
+        unique_pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 1 },
+          { "date" => (from - 2.days).to_s, "value" => 2 },
+          { "date" => (to + 1.day).to_s, "value" => 30 }
+        ],
+        pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 10 },
+          { "date" => (from - 2.days).to_s, "value" => 20 },
+          { "date" => (to + 1.day).to_s, "value" => 30 }
+        ]
+      })
+  end
+
+  def expect_metrics_for_each_date_to_be_correct(from, to)
+    date1 = (from - 1.day).to_s.last(5)
```

Comment:
By reading the code it is not obvious (to me, at least) what it does, or what is the shape of date1. Can you think of a better way to describe it?

Reply:
Done!